### PR TITLE
fix(polar): support borderRadius for polar bars #17980

### PR DESCRIPTION
**Pull request by Ovilia**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

support borderRadius for polar bars

### Fixed issues

#17980

### Before: What was the problem?

`itemStyle.borderRadius` is not supported for polar bars.

### After: How does it behave after the fixing?

`itemStyle.borderRadius` is supported like Sunburst or Pie charts.

## Document Info

One of the following should be checked.

- [ ] This PR doesn't relate to document changes
- [x] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

**Review by hexiaokang — APPROVED** [review]

File: src/chart/bar/BarView.ts [hunk 1]
```diff
@@ -981,7 +985,20 @@ function updateStyle(
     const style = data.getItemVisual(dataIndex, 'style');
 
     if (!isPolar) {
-        (el as Rect).setShape('r', itemModel.get(['itemStyle', 'borderRadius']) || 0);
+        const borderRadius = itemModel
+            .get(['itemStyle', 'borderRadius']) as number | number[] || 0;
+        (el as Rect).setShape('r', borderRadius);
+    }
+    else if (!seriesModel.get('roundCap')) {
+        const sector = el as Sector;
```
Comment (plainheart) [inline src/chart/bar/BarView.ts:993]: Unnecessary declaration for the variable `sector`. This line can be removed.

File: src/chart/bar/BarView.ts [hunk 2]
```diff
@@ -981,7 +985,20 @@ function updateStyle(
...
+        const sectorShape = (el as Sector).shape;
+        const cornerRadius = getSectorCornerRadius(
+            itemModel.getModel('itemStyle'),
+            sectorShape,
+            true
+        );
+        extend(sectorShape, cornerRadius);
+        sector.setShape(sectorShape);
```
Comment (plainheart) [inline src/chart/bar/BarView.ts:1001]: Use `(el as Sector).setShape(sectorShape)` instead.

**Review by plainheart — COMMENTED** [review]
And should we also support `borderRadius` for the background sector?
https://github.com/apache/echarts/blob/566743232967d95ea3a7b0ac688076f402db9ee7/src/chart/bar/BarView.ts#L246-L248

https://github.com/apache/echarts/blob/566743232967d95ea3a7b0ac688076f402db9ee7/src/chart/bar/BarView.ts#L337-L339

**Review by plainheart — APPROVED** [review]

**Comment by plainheart** [comment]
Merging latest master branch into this branch can solve the lint action error.

**Pull request merged**